Fix bug in WS error handling: ArgumentNullException

### Setup

Let's have AZP YML file with an **nonexistent** organization key:
```
- task: SonarCloudPrepare@1
 inputs:
    SonarCloud: 'SonarCloud PayBySquare.NET Service Connection'
    organization: 'ThisIsInvalidValue'
    scannerMode: 'MSBuild'
    projectKey: 'Something'
    projectName: 'Something'
```

### Error

It produces `ArgumentNullException`


```
2021-07-24T19:53:12.4529179Z ##[error]19:53:12.438  Failed to request and parse 'https://sonarcloud.io/api/qualityprofiles/search?project=Something&organization=ThisIsInvalidValue': Value cannot be null.
Parameter name: s
```

While the reported URL from SonarCloud
https://sonarcloud.io/api/qualityprofiles/search?project=Something&organization=ThisIsInvalidValue

returns clear error message that is not visible in the log
```
{"errors":[{"msg":"No organization with key \u0027ThisIsInvalidValue\u0027"}]}
```

### Full log
```
2021-07-24T19:53:10.5219836Z ##[section]Starting: SonarCloudPrepare
2021-07-24T19:53:10.5404829Z ==============================================================================
2021-07-24T19:53:10.5405723Z Task         : Prepare Analysis Configuration
2021-07-24T19:53:10.5406376Z Description  : Prepare SonarCloud analysis configuration
2021-07-24T19:53:10.5406773Z Version      : 1.20.0
2021-07-24T19:53:10.5407289Z Author       : sonarsource
2021-07-24T19:53:10.5407953Z Help         : Version: 1.20.0. [More Information](https://sonarcloud.io/documentation/analysis/scan/sonarscanner-for-azure-devops/)
2021-07-24T19:53:10.5408765Z ==============================================================================
2021-07-24T19:53:11.6940202Z [command]D:\a\_tasks\SonarCloudPrepare_14d9cde6-c1da-4d55-aa01-2965cd301255\1.20.0\classic-sonar-scanner-msbuild\SonarScanner.MSBuild.exe begin /k:Something /o:ThisIsInvalidValue
2021-07-24T19:53:11.8029673Z SonarScanner for MSBuild 5.2.1
2021-07-24T19:53:11.8037307Z Using the .NET Framework version of the Scanner for MSBuild
2021-07-24T19:53:11.8910853Z Pre-processing started.
2021-07-24T19:53:11.8944412Z Preparing working directories...
2021-07-24T19:53:12.0944275Z 19:53:12.093  Updating build integration targets...
2021-07-24T19:53:12.2988185Z 19:53:12.297  Fetching analysis configuration settings...
2021-07-24T19:53:12.4529179Z ##[error]19:53:12.438  Failed to request and parse 'https://sonarcloud.io/api/qualityprofiles/search?project=Something&organization=ThisIsInvalidValue': Value cannot be null.
Parameter name: s
2021-07-24T19:53:12.4550272Z 19:53:12.438  Failed to request and parse 'https://sonarcloud.io/api/qualityprofiles/search?project=Something&organization=ThisIsInvalidValue': Value cannot be null.
2021-07-24T19:53:12.4551520Z Parameter name: s
2021-07-24T19:53:12.4751840Z 
2021-07-24T19:53:12.4753780Z ##[error]Unhandled Exception:
2021-07-24T19:53:12.4895653Z ##[error]System.ArgumentNullException: Value cannot be null.
Parameter name: s
   at System.IO.StringReader..ctor(String s)
   at Newtonsoft.Json.Linq.JObject.Parse(String json, JsonLoadSettings settings)
   at SonarScanner.MSBuild.PreProcessor.SonarWebService.<>c__DisplayClass6_0.<<TryGetQualityProfile>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SonarScanner.MSBuild.PreProcessor.SonarWebService.<DoLogExceptions>d__17`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SonarScanner.MSBuild.PreProcessor.SonarWebService.<TryGetQualityProfile>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at SonarScanner.MSBuild.PreProcessor.TeamBuildPreProcessor.<FetchArgumentsAndRulesets>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SonarScanner.MSBuild.PreProcessor.TeamBuildPreProcessor.<DoExecute>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SonarScanner.MSBuild.PreProcessor.TeamBuildPreProcessor.<Execute>d__11.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SonarScanner.MSBuild.BootstrapperClass.<PreProcess>d__9.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SonarScanner.MSBuild.BootstrapperClass.<Execute>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SonarScanner.MSBuild.Program.<Execute>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SonarScanner.MSBuild.Program.<Execute>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SonarScanner.MSBuild.Program.<Main>d__2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at SonarScanner.MSBuild.Program.<Main>(String[] args)
2021-07-24T19:53:12.4916131Z System.ArgumentNullException: Value cannot be null.
2021-07-24T19:53:12.4918055Z Parameter name: s
2021-07-24T19:53:12.4919304Z    at System.IO.StringReader..ctor(String s)
2021-07-24T19:53:12.4920640Z    at Newtonsoft.Json.Linq.JObject.Parse(String json, JsonLoadSettings settings)
2021-07-24T19:53:12.4922138Z    at SonarScanner.MSBuild.PreProcessor.SonarWebService.<>c__DisplayClass6_0.<<TryGetQualityProfile>b__0>d.MoveNext()
2021-07-24T19:53:12.4923934Z --- End of stack trace from previous location where exception was thrown ---
2021-07-24T19:53:12.4925303Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2021-07-24T19:53:12.4926793Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2021-07-24T19:53:12.4928305Z    at SonarScanner.MSBuild.PreProcessor.SonarWebService.<DoLogExceptions>d__17`1.MoveNext()
2021-07-24T19:53:12.4929718Z --- End of stack trace from previous location where exception was thrown ---
2021-07-24T19:53:12.4931185Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2021-07-24T19:53:12.4932690Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2021-07-24T19:53:12.4934235Z    at SonarScanner.MSBuild.PreProcessor.SonarWebService.<TryGetQualityProfile>d__6.MoveNext()
2021-07-24T19:53:12.4935604Z --- End of stack trace from previous location where exception was thrown ---
2021-07-24T19:53:12.4936764Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2021-07-24T19:53:12.4938013Z    at SonarScanner.MSBuild.PreProcessor.TeamBuildPreProcessor.<FetchArgumentsAndRulesets>d__14.MoveNext()
2021-07-24T19:53:12.4939340Z --- End of stack trace from previous location where exception was thrown ---
2021-07-24T19:53:12.4940563Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2021-07-24T19:53:12.4941792Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2021-07-24T19:53:12.4943327Z    at SonarScanner.MSBuild.PreProcessor.TeamBuildPreProcessor.<DoExecute>d__12.MoveNext()
2021-07-24T19:53:12.4944617Z --- End of stack trace from previous location where exception was thrown ---
2021-07-24T19:53:12.4945835Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2021-07-24T19:53:12.4947946Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2021-07-24T19:53:12.4949287Z    at SonarScanner.MSBuild.PreProcessor.TeamBuildPreProcessor.<Execute>d__11.MoveNext()
2021-07-24T19:53:12.4950526Z --- End of stack trace from previous location where exception was thrown ---
2021-07-24T19:53:12.4951712Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2021-07-24T19:53:12.4954525Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2021-07-24T19:53:12.4955821Z    at SonarScanner.MSBuild.BootstrapperClass.<PreProcess>d__9.MoveNext()
2021-07-24T19:53:12.4957004Z --- End of stack trace from previous location where exception was thrown ---
2021-07-24T19:53:12.4958204Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2021-07-24T19:53:12.4959460Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2021-07-24T19:53:12.4960725Z    at SonarScanner.MSBuild.BootstrapperClass.<Execute>d__8.MoveNext()
2021-07-24T19:53:12.4961905Z --- End of stack trace from previous location where exception was thrown ---
2021-07-24T19:53:12.4963114Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2021-07-24T19:53:12.4964577Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2021-07-24T19:53:12.4966517Z    at SonarScanner.MSBuild.Program.<Execute>d__4.MoveNext()
2021-07-24T19:53:12.4967557Z --- End of stack trace from previous location where exception was thrown ---
2021-07-24T19:53:12.4968548Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2021-07-24T19:53:12.4969560Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2021-07-24T19:53:12.4970770Z    at SonarScanner.MSBuild.Program.<Execute>d__3.MoveNext()
2021-07-24T19:53:12.4971898Z --- End of stack trace from previous location where exception was thrown ---
2021-07-24T19:53:12.4972860Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2021-07-24T19:53:12.4973865Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2021-07-24T19:53:12.4974853Z    at SonarScanner.MSBuild.Program.<Main>d__2.MoveNext()
2021-07-24T19:53:12.4975759Z --- End of stack trace from previous location where exception was thrown ---
2021-07-24T19:53:12.4976714Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2021-07-24T19:53:12.4977712Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2021-07-24T19:53:12.4979591Z 
2021-07-24T19:53:12.5071442Z ##[error]The process 'D:\a\_tasks\SonarCloudPrepare_14d9cde6-c1da-4d55-aa01-2965cd301255\1.20.0\classic-sonar-scanner-msbuild\SonarScanner.MSBuild.exe' failed with exit code 3762504530
2021-07-24T19:53:12.5273788Z ##[section]Finishing: SonarCloudPrepare
```

--- a/Tests/SonarScanner.MSBuild.PreProcessor.Tests/PreProcessorTests.cs
+++ b/Tests/SonarScanner.MSBuild.PreProcessor.Tests/PreProcessorTests.cs
@@ -128,7 +128,6 @@ namespace SonarScanner.MSBuild.PreProcessor.Tests
         [TestMethod]
         public void PreProc_EndToEnd_SuccessCase_NoActiveRule()
         {
-
             // Arrange
             var workingDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
             var logger = new TestLogger();
@@ -449,6 +448,23 @@ namespace SonarScanner.MSBuild.PreProcessor.Tests
         }
 
         [TestMethod]
+        public void PreProc_HandleAnalysisException()
+        {
+            // Checks end-to-end behavior when AnalysisException is thrown inside FetchArgumentsAndRulesets
+            var workingDir = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var logger = new TestLogger();
+            var mockServer = new ThrowingSonarQubeServer();
+            var mockFactory = new MockObjectFactory(mockServer, new Mock<ITargetsInstaller>().Object, null);
+            using (new WorkingDirectoryScope(workingDir))
+            {
+                var preProcessor = new TeamBuildPreProcessor(mockFactory, logger);
+                var success = preProcessor.Execute(CreateValidArgs("key", "name", "1.0", "InvalidOrganization")).Result;    // Should not throw
+                success.Should().BeFalse("Expecting the pre-processing to fail");
+                mockServer.AnalysisExceptionThrown.Should().BeTrue();
+            }
+        }
+
+        [TestMethod]
         // Regression test for https://github.com/SonarSource/sonar-scanner-msbuild/issues/699
         public void PreProc_EndToEnd_Success_LocalSettingsAreUsedInSonarLintXML()
         {
@@ -619,5 +635,37 @@ namespace SonarScanner.MSBuild.PreProcessor.Tests
         }
 
         #endregion Checks
+
+        private class ThrowingSonarQubeServer : ISonarQubeServer
+        {
+            public bool AnalysisExceptionThrown { get; private set; }
+
+            public Task<IEnumerable<string>> GetAllLanguages() =>
+                Task.FromResult(new[] { TeamBuildPreProcessor.CSharpLanguage, TeamBuildPreProcessor.VBNetLanguage }.AsEnumerable());
+
+            public Task<IDictionary<string, string>> GetProperties(string projectKey, string projectBranch) =>
+                Task.FromResult<IDictionary<string, string>>(new Dictionary<string, string>());
+
+            public Task<IList<SonarRule>> GetRules(string qProfile) =>
+                throw new NotImplementedException();
+
+            public Task<Version> GetServerVersion() =>
+                Task.FromResult(new Version(8, 0));
+
+            public Task<bool> IsServerLicenseValid() =>
+                Task.FromResult(true);
+
+            public Task<bool> TryDownloadEmbeddedFile(string pluginKey, string embeddedFileName, string targetDirectory) =>
+                throw new NotImplementedException();
+
+            public Task<Tuple<bool, string>> TryGetQualityProfile(string projectKey, string projectBranch, string organization, string language)
+            {
+                AnalysisExceptionThrown = true;
+                throw new AnalysisException("This message and stacktrace should not propagate to the users");
+            }
+
+            public Task WarnIfSonarQubeVersionIsDeprecated() =>
+                Task.CompletedTask;
+        }
     }
 }

--- a/Tests/SonarScanner.MSBuild.PreProcessor.Tests/SonarWebServiceTest.cs
+++ b/Tests/SonarScanner.MSBuild.PreProcessor.Tests/SonarWebServiceTest.cs
@@ -227,6 +227,23 @@ namespace SonarScanner.MSBuild.PreProcessor.UnitTests
         }
 
         [TestMethod]
+        public void TryGetQualityProfile_SonarCloud_InvalidOrganizationKey()
+        {
+            const string serverUrl = "http://localhost:42424";
+            const string projectKey = "projectKey";
+            var mockDownloader = new Mock<IDownloader>(MockBehavior.Strict);
+            mockDownloader.Setup(x => x.Download($"{serverUrl}/api/server/version", false)).Returns(Task.FromResult("8.0.0.22548"));
+            mockDownloader.Setup(x => x.TryDownloadIfExists($"{serverUrl}/api/qualityprofiles/search?project={projectKey}&organization=ThisIsInvalidValue", false)).Returns(Task.FromResult(Tuple.Create(false, (string)null)));
+            mockDownloader.Setup(x => x.Download($"{serverUrl}/api/qualityprofiles/search?defaults=true&organization=ThisIsInvalidValue", false)).Returns(Task.FromResult<string>(null));    // SC returns 404, WebClientDownloader returns null
+            mockDownloader.Setup(x => x.Dispose());
+            using (var service = new SonarWebService(mockDownloader.Object, serverUrl, this.logger))
+            {
+                Action a = () => _ = service.TryGetQualityProfile("projectKey", null, "ThisIsInvalidValue", "cs").Result;
+                a.Should().Throw<ArgumentNullException>();
+            }
+        }
+
+        [TestMethod]
         public void TryGetQualityProfile64()
         {
             this.downloader.Pages["http://myhost:222/api/server/version"] = "6.4";

--- a/Tests/SonarScanner.MSBuild.PreProcessor.Tests/SonarWebServiceTest.cs
+++ b/Tests/SonarScanner.MSBuild.PreProcessor.Tests/SonarWebServiceTest.cs
@@ -239,7 +239,9 @@ namespace SonarScanner.MSBuild.PreProcessor.UnitTests
             using (var service = new SonarWebService(mockDownloader.Object, serverUrl, this.logger))
             {
                 Action a = () => _ = service.TryGetQualityProfile("projectKey", null, "ThisIsInvalidValue", "cs").Result;
-                a.Should().Throw<ArgumentNullException>();
+                a.Should().Throw<AggregateException>().WithMessage("One or more errors occurred.");
+                logger.AssertErrorLogged("Failed to request and parse 'http://localhost:42424/api/qualityprofiles/search?defaults=true&organization=ThisIsInvalidValue': Cannot download quality profile. Check scanner arguments and the reported URL for more information.");
+                logger.AssertErrorLogged("Failed to request and parse 'http://localhost:42424/api/qualityprofiles/search?project=projectKey&organization=ThisIsInvalidValue': Cannot download quality profile. Check scanner arguments and the reported URL for more information.");
             }
         }
 

--- a/src/SonarScanner.MSBuild.PreProcessor/Resources.Designer.cs
+++ b/src/SonarScanner.MSBuild.PreProcessor/Resources.Designer.cs
@@ -151,6 +151,15 @@ namespace SonarScanner.MSBuild.PreProcessor {
         }
         
         /// <summary>
+        ///   Looks up a localized string similar to Cannot download quality profile. Check scanner arguments and the reported URL for more information..
+        /// </summary>
+        internal static string ERROR_DownloadingQualityProfileFailed {
+            get {
+                return ResourceManager.GetString("ERROR_DownloadingQualityProfileFailed", resourceCulture);
+            }
+        }
+        
+        /// <summary>
         ///   Looks up a localized string similar to The following CheckId should not appear multiple times: {0}.
         /// </summary>
         internal static string ERROR_DuplicateCheckId {

--- a/src/SonarScanner.MSBuild.PreProcessor/Resources.resx
+++ b/src/SonarScanner.MSBuild.PreProcessor/Resources.resx
@@ -141,6 +141,9 @@
   <data name="ERROR_CmdLine_InvalidInstallTargetsValue" xml:space="preserve">
     <value>Invalid value for /install: {0}. Valid values are "true" or "false".</value>
   </data>
+  <data name="ERROR_DownloadingQualityProfileFailed" xml:space="preserve">
+    <value>Cannot download quality profile. Check scanner arguments and the reported URL for more information.</value>
+  </data>
   <data name="ERROR_DuplicateCheckId" xml:space="preserve">
     <value>The following CheckId should not appear multiple times: {0}</value>
   </data>

--- a/src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs
+++ b/src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs
@@ -74,7 +74,7 @@ namespace SonarScanner.MSBuild.PreProcessor
                         return ret == null
                             ? throw new AnalysisException(Resources.ERROR_DownloadingQualityProfileFailed)
                             : ret;
-                        }, ws);
+                    }, ws);
                 }
 
                 var json = JObject.Parse(contents);

--- a/src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs
+++ b/src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs
@@ -24,7 +24,6 @@ using System.IO;
 using System.Linq;
 using System.Net;
 using System.Threading.Tasks;
-using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 using SonarScanner.MSBuild.Common;
 using SonarScanner.MSBuild.PreProcessor.Roslyn.Model;
@@ -69,7 +68,13 @@ namespace SonarScanner.MSBuild.PreProcessor
                     ws = await AddOrganization(GetUrl("/api/qualityprofiles/search?defaults=true"), organization);
 
                     this.logger.LogDebug(Resources.MSG_FetchingQualityProfile, projectId, ws);
-                    contents = await this.downloader.Download(ws);
+                    contents = await DoLogExceptions(async() =>
+                    {
+                        var ret = await this.downloader.Download(ws);
+                        return ret == null
+                            ? throw new AnalysisException(Resources.ERROR_DownloadingQualityProfileFailed)
+                            : ret;
+                        }, ws);
                 }
 
                 var json = JObject.Parse(contents);

--- a/src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs
+++ b/src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs
@@ -68,13 +68,7 @@ namespace SonarScanner.MSBuild.PreProcessor
                     ws = await AddOrganization(GetUrl("/api/qualityprofiles/search?defaults=true"), organization);
 
                     this.logger.LogDebug(Resources.MSG_FetchingQualityProfile, projectId, ws);
-                    contents = await DoLogExceptions(async() =>
-                    {
-                        var ret = await this.downloader.Download(ws);
-                        return ret == null
-                            ? throw new AnalysisException(Resources.ERROR_DownloadingQualityProfileFailed)
-                            : ret;
-                    }, ws);
+                    contents = await DoLogExceptions(async () => await this.downloader.Download(ws) ?? throw new AnalysisException(Resources.ERROR_DownloadingQualityProfileFailed), ws);
                 }
 
                 var json = JObject.Parse(contents);

--- a/src/SonarScanner.MSBuild.PreProcessor/TeamBuildPreProcessor.cs
+++ b/src/SonarScanner.MSBuild.PreProcessor/TeamBuildPreProcessor.cs
@@ -123,7 +123,7 @@ namespace SonarScanner.MSBuild.PreProcessor
 
             var server = this.factory.CreateSonarQubeServer(localSettings);
 
-            //TODO: fail fast after release of 6.0
+            //TODO: fail fast after release of S4NET 6.0
             //Deprecation notice for SQ < 7.9
             await server.WarnIfSonarQubeVersionIsDeprecated();
 
@@ -229,6 +229,11 @@ namespace SonarScanner.MSBuild.PreProcessor
                         argumentsAndRuleSets.AnalyzersSettings.Add(analyzer);
                     }
                 }
+            }
+            catch (AnalysisException)
+            {
+                argumentsAndRuleSets.IsSuccess = false;
+                return argumentsAndRuleSets;
             }
             catch (WebException ex)
             {

--- a/src/SonarScanner.MSBuild.PreProcessor/WebClientDownloader.cs
+++ b/src/SonarScanner.MSBuild.PreProcessor/WebClientDownloader.cs
@@ -40,27 +40,23 @@ namespace SonarScanner.MSBuild.PreProcessor
             ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls;
 
             this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
-
             if (password == null)
             {
                 password = "";
             }
 
-            if (this.client == null)
+            if (clientCertPath != null && clientCertPassword != null) // password mandatory, as to use client cert in .jar it cannot be with empty password
             {
-                if (clientCertPath != null && clientCertPassword != null) // password mandatory, as to use client cert in .jar it cannot be with empty password
-                {
-                    var clientHandler = new HttpClientHandler();
-                    clientHandler.ClientCertificateOptions = ClientCertificateOption.Manual;
-                    clientHandler.ClientCertificates.Add(new X509Certificate2(clientCertPath, clientCertPassword));
-                    
-                    this.client = new HttpClient(clientHandler);
-                }
-                else
-                    this.client = new HttpClient();
-
-                this.client.DefaultRequestHeaders.Add(HttpRequestHeader.UserAgent.ToString(), $"ScannerMSBuild/{Utilities.ScannerVersion}");
+                var clientHandler = new HttpClientHandler { ClientCertificateOptions = ClientCertificateOption.Manual };
+                clientHandler.ClientCertificates.Add(new X509Certificate2(clientCertPath, clientCertPassword));
+                this.client = new HttpClient(clientHandler);
             }
+            else
+            {
+                this.client = new HttpClient();
+            }
+
+            this.client.DefaultRequestHeaders.Add(HttpRequestHeader.UserAgent.ToString(), $"ScannerMSBuild/{Utilities.ScannerVersion}");
 
             if (userName != null)
             {

